src/shell/shell_cmds.cpp

```diff
@@ -132,35 +132,48 @@ bool lookup_shell_cmd(std::string name, SHELL_Cmd &shell_cmd)
 	return true;
 }
 
-bool DOS_Shell::ExecuteConfigChange(const char* const cmd_in, const std::string args)
+bool DOS_Shell::ExecuteConfigChange(const char* const property_name, const std::string _args)
 {
 	assert(control);
 	const auto section_dosbox = static_cast<Section_prop*>(
 	        control->GetSection("dosbox"));
 
+	auto args = _args;
+	trim(args);
+
 	assert(section_dosbox);
 	if (!section_dosbox->Get_bool("shell_config_shortcuts")) {
 		return false;
 	}
 
-	Section* test = control->GetSectionFromProperty(cmd_in);
-	if (!test) {
+	Section* section = control->GetSectionFromProperty(property_name);
+	if (!section) {
 		return false;
 	}
 
 	if (args.empty()) {
-		std::string val = test->GetPropValue(cmd_in);
+		std::string val = section->GetPropValue(property_name);
 		if (val != NO_SUCH_PROPERTY) {
 			WriteOut("%s\n", val.c_str());
 		}
 		return true;
 	}
 
-	char newcom[1024];
-	safe_sprintf(newcom, "z:\\config -set %s %s%s", test->GetName(), cmd_in,
-	             args.c_str());
-	DoCommand(newcom);
+	if (args == "/?") {
+		const auto command = format_str("z:\\config -h %s %s",
+		                                section->GetName(),
+		                                property_name);
+
+		DoCommand(const_cast<char*>(command.c_str())); //-V2018
```

Comment:
This is UB since `DoCommand()` modifies the string. From a comment in my `basic_string.h` header: "This is a handle to internal data.  Do not modify or dire things may happen".

`.data()` apparently has an overload that allows mutation but I'm not sure how to use it: https://en.cppreference.com/w/cpp/string/basic_string/data

Another comment from the `.data()` method:

```cpp
      /**
       *  @brief  Return const pointer to contents.
       *
       *  This is a pointer to internal data.  It is undefined to modify
       *  the contents through the returned pointer. To get a pointer that
       *  allows modifying the contents use @c &str[0] instead,
       *  (or in C++17 the non-const @c str.data() overload).
      */
```

Reply:
Yeah the whole shell command execution stuff is a horrific mess of low-level C style string manipulation with mutation everywhere 🥶

Problem is you almost need to refactor everything to use `std::string` in a normal way. I'll see what I can do with minimal further refactors. Probably `DoCommand` would need to take a string view or `std::string` and _not_ mutate it in place... but it's used in so many places; that would be a big refactoring job.

The whole thing is a fucking mess. A much better would have been tokenise the command line into a vector of tokens, pass that arpund, then perhaphs maintain an index where we are at with processing the list of tokens, etc. instead of this direct string pointer manipulation hackery.

I hate C string handling.

Reply:
Reverted to use `safe_snprintf` and C strings. Converting to `std::string` would require a much deeper refactoring job so we don't just pass a `char *` around that everybody mutates... 🥶